dropshot/src/api_description.rs

```diff
@@ -252,11 +252,22 @@ impl ApiDescription {
         Ok(())
     }
 
+    /**
+     * Emit the OpenAPI definition describing this API.  Returns an
+     * OpenApiConfig which can be used to specify the contents of the definition
+     * and select an output format.
+     */
+    pub fn openapi<S1, S2>(&self, title: S1, version: S2) -> OpenApiConfig
+    where
+        S1: AsRef<str>,
+        S2: AsRef<str>,
+    {
+        OpenApiConfig::new(self, title.as_ref(), version.as_ref())
+    }
+
     /**
      * Emit the OpenAPI Spec document describing this API in its JSON form.
      */
-    // TODO: There's a bunch of error handling we need here such as checking
-    // for duplicate parameter names.
     pub fn print_openapi(
```

Comment:
Should we mark this deprecated?

Reply:
Yes, I think so!  I have added a note to the documentation and attached the deprecated attribute.